libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SalesforceNetReactBridge.java

```diff
@@ -152,32 +154,37 @@ private RestRequest prepareRestRequest(ReadableMap args) throws UnsupportedEncod
         Map<String, String> additionalHeaders = ReactBridgeHelper.toJavaStringStringMap(headerParams);
         Map<String, Object> queryParamsMap = ReactBridgeHelper.toJavaMap(queryParams);
         Map<String, Map<String, String>> fileParamsMap = ReactBridgeHelper.toJavaStringMapMap(fileParams);
-
         String urlParams = "";
         RequestBody requestBody = null;
-        if (method == RestRequest.RestMethod.DELETE || method == RestRequest.RestMethod.GET || method == RestRequest.RestMethod.HEAD) {
+        if (method == RestRequest.RestMethod.DELETE || method == RestRequest.RestMethod.GET
+                || method == RestRequest.RestMethod.HEAD) {
             urlParams = buildQueryString(queryParamsMap);
         } else {
             requestBody = buildRequestBody(queryParamsMap, fileParamsMap);
         }
-
         String separator = urlParams.isEmpty()
                 ? ""
                 : path.contains("?")
                     ? (path.endsWith("&") ? "" : "&")
                     : "?";
-
         return new RestRequest(method, endPoint + path + separator + urlParams, requestBody, additionalHeaders);
     }
 
     /**
      * Returns the RestClient instance being used by this bridge.
      *
+     * @param doesNotRequireAuth True - if an unauthenticated client should be used, False - otherwise.
      * @return RestClient instance.
      */
-    protected RestClient getRestClient() {
+    protected RestClient getRestClient(boolean doesNotRequireAuth) {
         final SalesforceReactActivity currentActivity = (SalesforceReactActivity) getCurrentActivity();
-        return currentActivity != null ? currentActivity.getRestClient() : null;
+        if (currentActivity == null) {
+            return null;
+        }
+        if (doesNotRequireAuth) {
+            return currentActivity.getClientManager().peekUnauthenticatedRestClient();
```

Comment:
Could you just change SalesforceReactActivity's getRestClient to take boolean doesNotRequireAuth so that you wouldn't have to expose a new method to get the client manager?

Reply:
It's public API - that's why I didn't change it ;-)

Reply:
I can make this API package-protected.

Reply:
Nope - they're in different packages.

Reply:
You could have 
```java
public RestClient getRestClient() {
    return client;
    // or: return getRestClient(false);
}

public RestClient getRestClient(boolean doesNotRequireAuth) {
    return doesNotRequireAuth ? clientManager. peekUnauthenticatedRestClient() : client;
 }
```

Reply:
I realized that we have public API for `ClientManager` in `SalesforceSDKManager`, so I removed the redundant method and rebuilding it. We don't need the new overridden API either.